kai/reactive_codeplanner/task_runner/analyzer_lsp/api.py

```diff
@@ -21,6 +23,40 @@ class AnalyzerRuleViolation(ValidationError):
     def __str__(self) -> str:
         return f"{self.__class__.__name__}<loc={self.file}:{self.line}:{self.column}, message={self.violation.description}>"
 
+    def background(self) -> str:
+        if self.parent is not None:
+            return self.oldest_ancestor().background()
+        if self.children:
+            return f"""You are a software developer who specializes in migrating from {" and ".join(self.sources)} to {" and ".join(self.targets)}
```

Comment:
I don't like having prompt creation on this object, but OTOH these objects know exactly what fields they have. Curious to hear people's thoughts

Reply:
I don't see a problem with using this object tbh, feels natural...just rename it to as_llm_context() or as_natural_language() or something

Reply:
Yeah maybe I'll just take out the instructions part of it and leave it as a summary, I think that would make it a lot cleaner